core/consensus/babe/primitives/src/lib.rs

```diff
@@ -49,15 +56,25 @@ pub type SlotNumber = u64;
 /// The weight of an authority.
 pub type Weight = u64;
 
+/// BABE epoch information
+#[derive(Decode, Encode, Default, PartialEq, Eq, Clone)]
+#[cfg_attr(any(feature = "std", test), derive(Debug))]
+pub struct Epoch {
```

Comment:
Maybe add the index here? This way when we're verifying or authoring a block we fetch the current epoch from runtime state instead of computing it ourselves (leading to potential inconsistencies when using `current_slots / slots_per_epoch`).

Reply:
yes

Reply:
@andresilva I had already done that before I saw your comment 😃